pgactivity/pg.py

```diff
@@ -169,6 +171,16 @@ def fetchall(
         with cursor(conn, mkrow, text_as_bytes) as cur:
             return cur.execute(query, args, prepare=True).fetchall()
 
+    def decode(value: bytes, pgenc: bytes, *, errors: str) -> str:
+        """Decode 'value' with PostgreSQL encoding 'pgenc' converted to Python encoding
+        name if available.
+        """
+        try:
+            pyenc = pg2pyenc(pgenc)
+        except NotSupportedError:
```

Comment:
This case is for psycopg2 ?

Reply:
Not sure I get your question... This function is for psycopg 3+, there is a similar function defined for psycopg2 below. If you ask about the try/except, no it's not for psycopg2 either, it is because some postgres encodings do not have an equivalent in Python and we thus fall back to utf-8; typically that's the case of `EUC_TW` which is covered in added tests.